processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
@akashdw such a benchmark would be too "far" from what will really happen in production. So that it's results are useless, it may show something speedup or slowdown, but in reality, on real > 2 GB files, the effect could be opposite to the benchmark.

Also currently "large mode" (v2) is aiming exceptional situations (not often), so it's "cold" even compared to GenericIndexed in general, which is not a particular hotspot (only `get()` method might be relatively hot).

That is why I don't want to make such benchmark.